Shopify/hydrogen
# Add CLI command to pull environment variables

### WHY are these changes introduced?

This is the next step in updating `hydrogen-cli` to better communicate and interact with Hydrogen storefronts created on Shopify. We want a seamless experience for developers when they are working on their storefront locally.

One of the biggest pain points for developers right now is ensuring that their environment variables are the same locally as they are on their Preview deployments to Oxygen. 

### WHAT is this pull request doing?

Adds a new `env pull` command to the Hydrogen CLI following the pattern of existing commands.

<details>
  <summary>Successful command run</summary>
  <img width="756" alt="Terminal screenshot showing successful output of running env pull" src="https://user-images.githubusercontent.com/836758/233485772-4ee8752e-8501-4bc3-a0bf-ba6786ded001.png">
</details>

<details>
  <summary>Error message when there is no linked storefront</summary>
<img width="643" alt="Terminal screenshot showing error message" src="https://user-images.githubusercontent.com/836758/233485024-9dfd597b-162b-41da-9f67-a768dd0b1bd7.png">
</details>

<details>
  <summary>Error message when Hydrogen storefront is missing in the Admin</summary>
  <img width="644" alt="Terminal screenshot showing error message" src="https://user-images.githubusercontent.com/836758/233485370-7c742e38-12b4-4a50-b953-d2427691163c.png">
</details>

As a starting point this command will *only* pull down environment variables for the `Preview` environment. We'll use this as a foundation and add more options to this command as we go.

### HOW to test your changes?

1. Pull the branch and navigate to the `cli` directory
1. Run `npm run build`
1. Run `export SHOPIFY_HYDROGEN_FLAG_PATH=../../templates/demo-store` so we're always targeting the demo store
1. Run `h2 env pull` (note: if `h2` is unavailable run `npx shopify hydrogen shortcut` first) to see a list of your storefronts
   - You should be prompted to provide a shop
   - You should then be prompted to authenticate with that shop
   - You should be prompted that you need to link to a storefront
1. Run `h2 link` to link to a specific storefront
1. Run `h2 env pull`
   - You should be prompted to confirm to overwrite the existing `.env` file
   - You should see all of your environment variables written to the `.env` file

#### Checklist
